scipy/signal/tests/test_signaltools.py

```diff
@@ -117,71 +136,76 @@ def test_input_swapping(self):
               [38 + 222j, 73 + 499j, 51 + 521j, 21 + 291j],
               [12 + 144j, 20 + 318j, 7 + 331j, 0 + 182j]]])
 
-        assert_array_equal(convolve(small, big, 'full'), out_array)
-        assert_array_equal(convolve(big, small, 'full'), out_array)
-        assert_array_equal(convolve(small, big, 'same'),
+        xp_assert_equal(convolve(small, big, 'full'), out_array)
+        xp_assert_equal(convolve(big, small, 'full'), out_array)
+        xp_assert_equal(convolve(small, big, 'same'),
                            out_array[1:3, 1:3, 1:3])
-        assert_array_equal(convolve(big, small, 'same'),
+        xp_assert_equal(convolve(big, small, 'same'),
                            out_array[0:3, 0:3, 0:3])
-        assert_array_equal(convolve(small, big, 'valid'),
+        xp_assert_equal(convolve(small, big, 'valid'),
                            out_array[1:3, 1:3, 1:3])
-        assert_array_equal(convolve(big, small, 'valid'),
+        xp_assert_equal(convolve(big, small, 'valid'),
                            out_array[1:3, 1:3, 1:3])
 
-    def test_invalid_params(self):
-        a = [3, 4, 5]
-        b = [1, 2, 3]
+    @array_api_compatible
+    def test_invalid_params(self, xp):
+        a = xp.array([3, 4, 5])
+        b = xp.array([1, 2, 3])
         assert_raises(ValueError, convolve, a, b, mode='spam')
         assert_raises(ValueError, convolve, a, b, mode='eggs', method='fft')
         assert_raises(ValueError, convolve, a, b, mode='ham', method='direct')
         assert_raises(ValueError, convolve, a, b, mode='full', method='bacon')
         assert_raises(ValueError, convolve, a, b, mode='same', method='bacon')
 
-
-class TestConvolve(_TestConvolve):
-
-    def test_valid_mode2(self):
+    @skip_xp_backends("jax.numpy", reason="dtypes do not match")
+    @array_api_compatible
+    def test_valid_mode2(self, xp):
         # See gh-5897
-        a = [1, 2, 3, 6, 5, 3]
-        b = [2, 3, 4, 5, 3, 4, 2, 2, 1]
-        expected = [70, 78, 73, 65]
+        a = xp.array([1, 2, 3, 6, 5, 3])
+        b = xp.array([2, 3, 4, 5, 3, 4, 2, 2, 1])
+        expected = xp.array([70, 78, 73, 65])
 
         out = convolve(a, b, 'valid')
-        assert_array_equal(out, expected)
+        xp_assert_equal(out, expected)
 
         out = convolve(b, a, 'valid')
-        assert_array_equal(out, expected)
+        xp_assert_equal(out, expected)
 
-        a = [1 + 5j, 2 - 1j, 3 + 0j]
-        b = [2 - 3j, 1 + 0j]
-        expected = [2 - 3j, 8 - 10j]
+        a = xp.array([1 + 5j, 2 - 1j, 3 + 0j])
```

Comment:
`xp.array` is not standard

```suggestion
        a = xp.asarray([1 + 5j, 2 - 1j, 3 + 0j])
```